cli/src/pcluster/config/cluster_config.py

```diff
@@ -658,6 +658,45 @@ def __init__(self, secured: bool = None, **kwargs):
         self.secured = Resource.init_param(secured, default=True)
 
 
+class DirectoryService(Resource):
+    """Represent DirectoryService configuration."""
+
+    def __init__(
+        self,
+        domain_name: str = None,
+        domain_addr: str = None,
+        password_secret_arn: str = None,
+        domain_read_only_user: str = None,
+        ldap_tls_ca_cert: str = None,
+        ldap_tls_req_cert: str = None,
+        ldap_access_filter: str = None,
+        generate_ssh_keys_for_users: bool = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.domain_name = Resource.init_param(domain_name)
+        self.domain_addr = Resource.init_param(domain_addr)
+        self.password_secret_arn = Resource.init_param(password_secret_arn)
+        self.domain_read_only_user = Resource.init_param(domain_read_only_user)
+        self.ldap_tls_ca_cert = Resource.init_param(ldap_tls_ca_cert)
+        self.ldap_tls_req_cert = Resource.init_param(ldap_tls_req_cert, default="hard")
+        self.ldap_access_filter = Resource.init_param(ldap_access_filter)
+        self.generate_ssh_keys_for_users = Resource.init_param(generate_ssh_keys_for_users, default=True)
+
+    def get_dna_json(self):
+        """Return the object to be written to dna.json."""
+        return {
```

Comment:
this is very similar to the self.__dict__ output except for the latest key, then the "dna_json" word doesn't have any sense in this file.

I think we can avoid this function and just have the "get_dna_json" in the cluster_stack.py that uses the keys and values from `__dict__.items()` appyling the `lower()`function only for the latest field.

Reply:
An alternative is to reuse the get_directory_service_user_data_env and convert the key of the output dictionary from CamelCase to snake_case

Reply:
Done with adjustment

`__dict__.items()` is not reliable because it retrieves other internal parameters